### ISSUE TITLE
[Bug] Explorer hangs if network share is inaccessible

### ISSUE THREAD

**Issue by tryallthethings** [nextcloud/desktop#4304]
### Expected behavior
Nextcloud Desktop sync client should just mark the folder sync as error if the network share is not reachable at the moment.

### Actual behavior
tldr: Nextcloud Desktop sync client causes explorer.exe and related processes to hang or have a huge delay in response.

I got 3 Folder Syncs configured in my desktop sync client. One of those was a Samba network share on an Ubuntu server. Last week I retired that server. Naturally, the share wasn't available anymore after that. This caused the Nextcloud client to behave weirdly. When it was started during after logging into Windows or via Start menu, it almost never got to a responsive state. Neither right nor left click worked on the icon in the taskbar to change the settings. Sometimes (after right-clicking and waiting for 10-15 minutes), I could select settings. The settings window would open and immediately be unresponsive again. So I was never able to remove or change the folder sync location. Additionally, while Nextcloud was running, Windows Explorer wouldn't open at all or freeze (application is not responding) during copy / pasting and especially while opening the context menu on files inside and outside of Nextcloud sync folders. The start menu was showing similar issues (late response / freezes). As soon as I "killed" the nextcloud.exe process via task manager, my system would work as normal. (In case it's relevant: I tried removing and reinstalling the client with the same result. Sync configuration was kept between installations) In the end, I had to edit nextcloud.cfg manually to change the folder sync to the new path. I can reproduce the issue by switching the server hostname in the nextcloud.cfg between the old (offline) and new server share.

### Steps to reproduce
1. Have a folder sync configured to a samba network share
2. Switch off server
3. Reboot client pc

### Client configuration
Client version: 3.4.2
Operating system: Windows 10 Pro 21H1
OS language: en-us
Installation path of client: default (C:\Program Files\Nextcloud)

### Server configuration
Nextcloud version: 22.2.5

### Logs
1. Client logfile: 
As I couldn't upload the debug archive as a single file (~34MB in size) I split it into 2:
[nextcloud_1.zip](https://github.com/nextcloud/desktop/files/8103309/nextcloud_1.zip)
[nextcloud_2.zip](https://github.com/nextcloud/desktop/files/8103310/nextcloud_2.zip)

**Comment by tryallthethings** [comment]
Issue still present in version 3.4.4.

**Comment by tryallthethings** [comment]
Could someone please at least acknowledge this bug, so it doesn't go stale every 30 days?

**Comment by Aldaris1985** [comment]
Same issue for me too.  The hanging does not last that long for me (maybe ~30 seconds) but it is still annoying. Also once a folder is loaded in the explorer it does not hang again. Interestingly this issue applies to folders that are actually not synced with Nextcloud  like "Desktop" or "Downloads". On an additional note: this issue also arises even if the folder synced over network is set to "Paused". 

> Could someone please at least acknowledge this bug, so it doesn't go stale every 30 days?

**Comment by Aldaris1985** [comment]
> Reporting bugs for Nextcloud sure is annoying.

@FlexW: Can you please remove the stale flag from this bug report? We have at least two users with this issue who confirmed.

**Comment by Aldaris1985** [comment]
> This bug report is getting automatically closed due to no answer since the issue has been staled. Thank you!

The bug remains... please keep it open!

**Comment by Aldaris1985** [comment]
Issue remains with 3.6.0

**Comment by joneuhauser** [comment]
I have the same issue, Nextcloud client 3.6.0. The symptoms are identical (Nextcloud is almost never responsive, Explorer freezes up, right click hangs even on non-Nextcloud files, and after closing Nextcloud, everything is working again). However, I don't have any inaccessible share, just one, perfectly accessible (at least in the browser) nextcloud drive.

**Comment by MingLin-home** [comment]
I have the same issue with 3.6.1 and 3.5.X.

**Comment by Bizikneidi** [comment]
Same problem here on version 3.6.1, after opening the explorer it just freezes for ~15 seconds, after this initial freeze it just behaves normally, even after closing and reopening the explorer. Only after a restart or after waking from sleep the issue comes up again, really annoying. The problems started after switching to virtual file support and persisted since then.

**Comment by Aldaris1985** [comment]
Issue remains with 3.6.2

**Comment by zBeeble42** [comment]
I believe I'm affected by this issue, too.

**Comment by Aldaris1985** [comment]
Issue remains with 3.6.4 ...

**Comment by Aldaris1985** [comment]
Issue remains with 3.7.1

**Comment by tryallthethings** [comment]
Issue remains with 3.7.3

**Comment by Xansala** [comment]
Issue remains with 3.8.0

**Comment by Xansala** [comment]
Issue remains with 3.8.1

**Comment by Xansala** [comment]
Remains with 3.8.2

**Comment by nihues** [comment]
Started getting 3.9.0 after moving to 27 branch... (and changed dns configuration), so not sure exactly where is the cause, but it's annoying as hell.

**Comment by Xansala** [comment]
Remains in 3.9.1

**Comment by tryallthethings** [comment]
I wasn't able to reproduce with version 3.9.3. Can you please check if you can? @Xansala @nihues @Bizikneidi

**Comment by Xansala** [comment]
> I wasn't able to reproduce with version 3.9.3. Can you please check if you can? @Xansala @nihues @Bizikneidi
> 
> Small update: While my Explorer seems unaffected so far, the Nextcloud itself is unresponsive and opening the settings from the taskbar takes minutes.

Unfortunately the issue remains for me. It is a bit different now. When booting the PC and the network location is not reachable, then it seems to work. But when I open Firefox, Print a page and save as PDF, the explorer still hangs with 3.9.3.

**Comment by Xansala** [comment]
Remains with 3.10.1

**Comment by pspeter** [comment]
I experience the same or at least very similar issues right now. However, my Nextcloud Server is still available and reachable (I can log into it using the browser).  In fact, I believe the issues started when I added a semi-big chunk of files to my local Nextcloud folder to get synced (it did not start immediately, only after some time, maybe after a reboot).

Clicking the Nextcloud tray icon (right or left) does nothing. Hovering over it displays the tooltip "Nextcloud: Sync is running".

My Windows Explorer is completely messed up:
 - I cannot open it via the icon in the task bar (however, `Win+E` still works)
 - right clicking any icon inside the explorer freezes the explorer and desktop until you restart the explorer process

Closing the nextcloud client via task manager fixes the explorer.

Looking at the task manager, it doesn't seem like nextcloud is actually syncing anything, the network traffic is 0%.
![image](https://github.com/nextcloud/desktop/assets/13824347/7b3a6031-72d9-49fd-a321-8e4aab1e2b3c)

In the logs, there's lots of warnings that look like this:

```
2024-01-11 22:26:13:442 [ warning nextcloud.sync.propagator.directory C:\Users\User\AppData\Local\Temp\windows-19197\client-building\desktop\src\libsync\owncloudpropagator.cpp:1251 ]:	Useless task found for file "path/to/file" instruction CSyncEnums::CSYNC_INSTRUCTION_SYNC
```

and a few:

```
2024-01-11 22:26:13:442 [ warning nextcloud.gui.activity C:\Users\User\AppData\Local\Temp\windows-19197\client-building\desktop\src\gui\tray\usermodel.cpp:840 ]:	Item  "other/file"  retrieved resulted in error  "Ignored because of the \"choose what to sync\" blacklist"
```

At the end, there's a few of these info logs before it completely stops and does not continue:

```
2024-01-11 22:26:13:454 [ info nextcloud.sync.propagator.bulkupload C:\Users\User\AppData\Local\Temp\windows-19197\client-building\desktop\src\libsync\bulkpropagatorjob.cpp:189 ]:	"different/file/path" transmission checksum "64fd03305a37a12d0e3d9360c81fd3b9" "C:/Users/local/different/file/path"
```

**Comment by pspeter** [comment]
I experienced the same or at least very similar issues. However, my Nextcloud Server was still available and reachable (I can log into it using the browser).  In fact, I believe the issues started when I added a semi-big chunk of files to my local Nextcloud folder to get synced (it did not start immediately, only after some time, maybe after a reboot).

Clicking the Nextcloud tray icon (right or left) does nothing. Hovering over it displays the tooltip "Nextcloud: Sync is running".

My Windows Explorer is completely messed up:
 - I cannot open it via the icon in the task bar (however, `Win+E` still works)
 - right clicking any icon inside the explorer freezes the explorer and desktop until you restart the explorer process

Closing the nextcloud client via task manager fixes the explorer.

Looking at the task manager, it doesn't seem like nextcloud is actually syncing anything, the network traffic is 0%.
![image](https://github.com/nextcloud/desktop/assets/13824347/7b3a6031-72d9-49fd-a321-8e4aab1e2b3c)

In the logs, there's lots of warnings that look like this:

```
2024-01-11 22:26:13:442 [ warning nextcloud.sync.propagator.directory C:\Users\User\AppData\Local\Temp\windows-19197\client-building\desktop\src\libsync\owncloudpropagator.cpp:1251 ]:	Useless task found for file "path/to/file" instruction CSyncEnums::CSYNC_INSTRUCTION_SYNC
```

and a few:

```
2024-01-11 22:26:13:442 [ warning nextcloud.gui.activity C:\Users\User\AppData\Local\Temp\windows-19197\client-building\desktop\src\gui\tray\usermodel.cpp:840 ]:	Item  "other/file"  retrieved resulted in error  "Ignored because of the \"choose what to sync\" blacklist"
```

At the end, there's a few of these info logs before it completely stops and does not continue:

```
2024-01-11 22:26:13:454 [ info nextcloud.sync.propagator.bulkupload C:\Users\User\AppData\Local\Temp\windows-19197\client-building\desktop\src\libsync\bulkpropagatorjob.cpp:189 ]:	"different/file/path" transmission checksum "64fd03305a37a12d0e3d9360c81fd3b9" "C:/Users/local/different/file/path"
```

I managed to fix it by deleting the folder mentioned in the bulkupload line at the end.

**Comment by Xansala** [comment]
Issue remains with 3.13.2

**Comment by joshtrichards** [comment]
@tryallthethings Your original report says "network share". Are you referring to a server-side Nextcloud *External Storage* mount?

**Comment by tryallthethings** [comment]
@joshtrichards My original network post said "Samba network share". Or call it a windows share / SMB protocol if you like.

**Comment by joshtrichards** [comment]
Yes, but where/how is the network share associated with the client? Is it an [SMB-based *External Storage* mount](https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/external_storage_configuration_gui.html) on the Nextcloud Server? Or are you referring to an SMB mount you have on your client device?

**Comment by Xansala** [comment]
@joshtrichards I let @tryallthethings  answer as well for his situation ;) . For me it is a SMB mount I have on the client device that I sync only once a while for a backup into the cloud. To be concrete: it is a picture folder I have on my NAS for which I have a backup in the cloud on my VPS and I use nextcloud to sync it. However, the NAS is not always running, so when the NAS is not connected, the client is hanging.

**Comment by tryallthethings** [comment]
My apologies, I assumed my initial post was clear on that matter, and I was under the impression that folder sync is only available in the desktop client. 

> I got 3 Folder Syncs configured in my desktop sync client. One of those was a Samba network share on an Ubuntu server.



> ### Steps to reproduce
> 1. Have a folder sync configured to a samba network share
> 2. Switch off server
> 3. Reboot client pc

So my setup at the time I raised the issue was very similar to what @Xansala is using. I just tried to reproduce the issue with the latest desktop client (3.13.3). While I still saw some issues with the Windows Explorer behaving odd / slow, I couldn't get it to freeze up completely as it did before. So I'd assume the issue is still present. At least the client stays mostly responsive, and I can get into the settings or show the status

![image](https://github.com/user-attachments/assets/e3346549-0a10-4850-9642-ad6f4f606d03)

**Comment by mgallien** [comment]
@tryallthethings thanks for the report
sorry for the delay
while it may still be possible to run the client synchronizing using a network folder, this is really not a good idea
you will definitely have big performance issues (in addition to what you report)
we do not plan to work on fixing the issues (if that was even possible from a technical perspective) and would rather have the client refuses to use a network folder as the place to synchronize files and folders
as an example using SQLite on a remote SMB share is very slow and not something we can fix. This would result in freezes of the client and bad user experience
closing this as not planned work

**Comment by tryallthethings** [comment]
@mgallien 
I see this was closed as “not planned,” but the problem still seems relevant: when a local sync target points to a network share that’s offline, explorer.exe (and context menus) can stall until the share times out. That freezes the whole desktop and can't be the intended behavior here. Besides this issue, I never noticed any issues with using a network share. But I'm only syncing a couple of PDFs from a scanner share. If this isn't and won't be supported, it would be best to not allow UNC paths as a sync folder it in the first place. Or at least give the users a warning when they use a UNC path. But I'd prefer this being fixed. A safeguard check if the path is available might already remediate the issue. I consider an issue that could cause the whole system to be unusable / unstable something that I would definitely look into.

I’m happy to retest on the latest client and provide fresh logs if this can be reconsidered. Also noting a newer report with similar symptoms on Windows: #6422.
